tensorboard/plugins/text_v2/text_demo_v2.py

```diff
@@ -0,0 +1,146 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from absl import app
+from absl import logging
+import tensorflow as tf
+from tensorboard.plugins.text.summary_v2 import text
+
+tf.compat.v1.enable_v2_behavior()
+
+# Directory into which to write tensorboard data.
+LOGDIR = "/tmp/text_demo"
+
+# Number of steps for which to write data.
+STEPS = 16
+
+
+def simple_example(step):
+    # Text summaries log arbitrary text. This can be encoded with ASCII or
+    # UTF-8. Here's a simple example, wherein we greet the user on each
+    # step:
+    step_string = tf.as_string(step)
+    greeting = tf.strings.join(["Hello from step ", step_string, "!"])
+    text("greeting", greeting, step)
+
+
+def simple_example_with_pagination(step):
+    # Text summaries can be grouped by Tag (the first argument to the `text`
+    # summary's API).  Unique tags will have their own dropdown in the frontend,
+    # unless those tags are sub-tags of another tag.  To create a subtag name,
+    # one would input MainTagName/SubTagName as the input argument to the `text`
+    # summary API.  Subtags will be displayed within the main tag's dropdown,
+    # and a pager will appear if there are more than 12 subtags.  This test
+    # displays that UI feature by creating 36 subtags within the main tag:
+    # "simple_example_with_pagination"
+    for i in range(36):
+        step_string = tf.as_string(step)
+        greeting = tf.strings.join(["Hello from step ", step_string, "!"])
+        text("Card" + str(i + 1) + "/greeting", greeting, step)
+
+
+def markdown_table(step):
+    # The text summary can also contain Markdown, including Markdown
+    # tables. Markdown tables look like this:
+    #
+    #     | hello | there |
+    #     |-------|-------|
+    #     | this  | is    |
+    #     | a     | table |
+    #
+    # The leading and trailing pipes in each row are optional, and the text
+    # doesn't actually have to be neatly aligned, so we can create these
+    # pretty easily. Let's do so.
+    header_row = "Pounds of chocolate | Happiness"
+    chocolate = tf.range(step)
+    happiness = tf.square(chocolate + 1)
+    chocolate_column = tf.as_string(chocolate)
+    happiness_column = tf.as_string(happiness)
+    table_rows = tf.strings.join([chocolate_column, " | ", happiness_column])
+    table_body = tf.strings.reduce_join(inputs=table_rows, separator="\n")
+    table = tf.strings.join([header_row, "---|---", table_body], separator="\n")
```

Comment:
While I don't have strong feelings, if you think it is easier to form a string in Python, then use `tf.constant`, feel free to refactor this block.

(btw, while this method is readable, the `higher_order_tensors` one look more challenging)

Reply:
I think for forming markdown tables, the current way is more clear.  But I do agree it gets a little more funky with the higher_order_tensors test, not sure if converting to python string formatting would help much tho.